### PR TITLE
Fix process kill on Mac OS X Mavericks

### PR DESCRIPTION
**Pull request by Aigeruth**
**The problem**: `guard-go` gets the `pid` of the `go run` command and not the pid of the running application. It sends the `KILL` signal to the `go run` process and exits, but the child process remains running.

**The solution**: change the application starting process to `go build` + `./binary`.

**Host:** Mac OS X Mavericks

``` bash
$ go version
go version go1.3 darwin/amd64
```

**This pull request is based on #10.**

### CODE REVIEW

**Comment by vcastellm** [comment]
I'm not sure if I like this solution, did you tested it?

**Comment by Aigeruth** [comment]
Yes, I tested it and it works for me. I use it for a [go-restful](https://github.com/emicklei/go-restful) project. But I agree on that it is not the most beautiful solution.

As I see, #8 can be related to this PR.

File: lib/guard/go/runner.rb [hunk 1]
```diff
@@ -46,9 +46,10 @@ def run_go_command!
         @proc = ChildProcess.build(@options[:cmd], "test")
       else
         if @options[:build_only]
-          @proc = ChildProcess.build(@options[:cmd], "build")
+          @proc = ChildProcess.build options[:cmd], "build", @options[:server]
```
Comment (liamdawson) [inline lib/guard/go/runner.rb]: Did you mean to remove that @ from the first options?

Comment (Aigeruth) [inline lib/guard/go/runner.rb]: You are right. Thanks!

**Comment by Aigeruth** [comment]
I've updated this PR.

**Pull request merged**